File: src/components/layout.js
/**
 * Layout component that queries for data
 * with Gatsby's useStaticQuery component
 *
 * See: https://www.gatsbyjs.org/docs/use-static-query/
 */

import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'

import Header from './header'
import Footer from './footer'
// import "./layout.css"

const Layout = ({ location, children }) => {
  const data = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          menuLinks {
            name
            link
          }
        }
      }
    }
  `)
  return (
    <>
      <Header
        menuLinks={data.site.siteMetadata.menuLinks}
        location={location}
      />
      <div className="container mx-auto md:max-w-screen-md">
        <main>{children}</main>
        <Footer />
      </div>
    </>
  )
}

export default Layout

File: website/src/main/kotlin/components/Layout.kt
package dev.luisramos.website.components

import kotlinx.html.FlowContent
import kotlinx.html.div
import kotlinx.html.main

fun FlowContent.Layout(location: String, content: FlowContent.() -> Unit) {
    Header(currLink = location)
    div(classes = "container mx-auto md:max-w-screen-md") {
        main {
            content()
        }
        Footer()
    }
}